jax/_src/interpreters/mlir.py

```diff
@@ -410,17 +404,49 @@ class ShapePolyLoweringState:
   # from an inner call to a polymorphic Exported.
   uses_dim_vars: bool
 
-  def __init__(self, dim_vars: tuple[str, ...],
-               lowering_platforms: tuple[str, ...]):
-    self.lowering_platforms = lowering_platforms
+  # If the first dimension variable is a platform index argument
+  has_platform_index_argument: bool
+
+  def __init__(self,
+               dim_vars: tuple[str, ...],
+               lowering_platforms: tuple[str, ...] | None):
     self.uses_dim_vars = (len(dim_vars) > 0)
-    if len(lowering_platforms) > 1:
+    if lowering_platforms is not None and len(lowering_platforms) > 1:
       dim_vars = ("platform_index_",) + tuple(dim_vars)
+      self.has_platform_index_argument = True
+    else:
+      self.has_platform_index_argument = False
     self.dim_vars = dim_vars
 
-  @property
-  def has_platform_index_argument(self):
-    return len(self.lowering_platforms) > 1
+
+dataclasses.dataclass(frozen=True)
+class LoweringParameters:
+  # A mapping between primitives and user-defined LoweringRules.
+  # When lowering a primitive, give priorioty to the rule in this map over
+  # existing Jax rules.
+  override_lowering_rules: tuple[tuple[core.Primitive, LoweringRule]] | None = None
+
+  # The lowering_platform, in case we do cross-platform lowering.
+  # One of 'cpu', 'cuda', 'rocm', 'tpu'.
+  # If None, use the default JAX mechanisms to pick the lowering platform.
+  # This is currently used for export and jax2tf.
+  lowering_platform: str | None
+
+  # The current lowering platforms, a non-empty tuple containing some of
+  # 'cpu', 'cuda', 'rocm', 'tpu'.
+  lowering_platforms: tuple[str, ...] | None
+
+  def __init__(self, *,
```

Comment:
It is unusual to define a custom `__init__` when writing a dataclass. I think you could remove it, if you eliminate the redundancy (and the need for validation) of `lowering_platform{,s}`.